### PR TITLE
Add model reference support to library and command line utility.   

### PR DESCRIPTION
**Pull request by garethj-msft**
Add model reference support to library and command line utility.   
Added basic model reference support to the library.
Added similar support to the commandline utility for local files only.
Also added a lot more conversion options flags to the command line.
Added library tests to match.

### CODE REVIEW

File: src/Microsoft.OpenApi.OData.Reader/OData/ODataConstants.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+{
+    internal static class ODataConstants
+    {
+        /// <summary>
+        /// Namespaces used in standard included models.
+        /// </summary>
+        public static IList<string> StandardNamespaces = new List<string>
+        {
+            "Org.OData.",
+            "Edm",
```
Comment (xuzhg) [inline src/Microsoft.OpenApi.OData.Reader/OData/ODataConstants.cs:22]: > "Edm", [](start = 11, length = 7)

others have "." at the end. so how about "Edm."?

File: src/Microsoft.OpenApi.OData.Reader/Edm/EdmModelExtensions.cs [hunk 2]
```diff
@@ -197,5 +197,28 @@ public static bool IsOperationImportOverload(this IEdmModel model, IEdmOperation
             return model.EntityContainer.OperationImports()
                 .Where(o => o.Operation.IsBound == operationImport.Operation.IsBound && o.Name == operationImport.Name).Count() > 1;
         }
+
+        /// <summary>
+        /// Get all of the elements in the model and its referenced models.
+        /// </summary>
+        /// <returns>All the elements.</returns>
+        public static IEnumerable<IEdmSchemaElement> GetAllElements(this IEdmModel model)
```
Comment (xuzhg) [inline src/Microsoft.OpenApi.OData.Reader/Edm/EdmModelExtensions.cs:205]: >GetAllElements(this IEdmModel model) [](start = 53, length = 36)

We should consider the size of the output.
The output of Open API only for Microsoft.Graph CSDL is very big.

File: src/OoasUtil/ComLineProcesser.cs [hunk 3]
```diff
@@ -132,6 +170,46 @@ public bool Process()
                             i++;
                             break;
 
+                        case "--derivedtypesreferencesforresponses":
```
Comment (xuzhg) [inline src/OoasUtil/ComLineProcesser.cs:173]: >derivedtypesreferencesforresponses [](start = 32, length = 34)

Shall we use Pascal format for mulitiple words?

File: test/Microsoft.OpenAPI.OData.Reader.Tests/Edm/EdmModelExtensionsTests.cs [hunk 4]
```diff
@@ -55,5 +59,23 @@ public void FindAllBaseTypesReturnsCorrect()
             Assert.Equal(2, baseTypes.Count());
             Assert.Equal(new[] { subEntityType, baseEntityType }, baseTypes);
         }
+
+
```
Comment (xuzhg) [inline test/Microsoft.OpenAPI.OData.Reader.Tests/Edm/EdmModelExtensionsTests.cs:63]: remove unused line

File: test/Microsoft.OpenAPI.OData.Reader.Tests/Edm/EdmModelExtensionsTests.cs [hunk 5]
```diff
@@ -55,5 +59,23 @@ public void FindAllBaseTypesReturnsCorrect()
...
+            IEdmModel model = EdmModelHelper.InheritanceEdmModelAcrossReferences;
+
+            // Act
+            var elements = model.GetAllElements();
+
+            // Assert
+            Assert.Collection(elements, 
+                e => Assert.Equal("Customer", e.Name),
+                e => Assert.Equal("Default", e.Name),
+            e => Assert.Equal("CustomerBase", e.Name));
```
Comment (xuzhg) [inline test/Microsoft.OpenAPI.OData.Reader.Tests/Edm/EdmModelExtensionsTests.cs:77]: alignment and remove the unused line at the end

File: test/Microsoft.OpenAPI.OData.Reader.Tests/Generator/OpenApiSchemaGeneratorTests.cs [hunk 6]
```diff
@@ -488,7 +534,7 @@ public void CreatePropertySchemaForNullableEnumPropertyReturnSchema(OpenApiSpecV
         public void GetDerivedTypesReferenceSchemaReturnsDerivedTypesReferencesInSchemaIfExist()
         {
             // Arrange
-            IEdmModel edmModel = EdmModelHelper.GraphBetaModel;
+            IEdmModel edmModel = OData.Tests.EdmModelHelper.GraphBetaModel;
```
Comment (xuzhg) [inline test/Microsoft.OpenAPI.OData.Reader.Tests/Generator/OpenApiSchemaGeneratorTests.cs:537]: > OData.Tests. [](start = 32, length = 13)

the above change use "Test." prefix, so, shall we use the same here?
Or do we remove them?

**Review by xuzhg — APPROVED** [review]
:shipit:

File: src/Microsoft.OpenApi.OData.Reader/OData/ODataConstants.cs (hunk 1 above)
Comment (garethj-msft) [reply src/Microsoft.OpenApi.OData.Reader/OData/ODataConstants.cs:22]: Th eothers do, but for the main Edm namesapce that is referenced by standard, it never does.  e.g. integer and string are just Edm.integer

File: src/Microsoft.OpenApi.OData.Reader/Edm/EdmModelExtensions.cs (hunk 2 above)
Comment (garethj-msft) [reply src/Microsoft.OpenApi.OData.Reader/Edm/EdmModelExtensions.cs:205]: I mean that is the whole point of this PR - to allow you to make a big OAS file that is combined from multiple smaller CSDL files.   If it is too big, then it is too big, but most OAS tools don't support file composition either.

File: src/OoasUtil/ComLineProcesser.cs (hunk 3 above)
Comment (garethj-msft) [reply src/OoasUtil/ComLineProcesser.cs:173]: I was following the convention of the existing 'specversion' which does not use PascalCase.

File: test/Microsoft.OpenAPI.OData.Reader.Tests/Generator/OpenApiSchemaGeneratorTests.cs (hunk 6 above)
Comment (garethj-msft) [reply test/Microsoft.OpenAPI.OData.Reader.Tests/Generator/OpenApiSchemaGeneratorTests.cs:537]: Remove - this was just a hangover from a namespace change I was fiddling with and forgot to remove.

File: test/Microsoft.OpenAPI.OData.Reader.Tests/Edm/EdmModelExtensionsTests.cs (hunk 5 above)
Comment (garethj-msft) [reply test/Microsoft.OpenAPI.OData.Reader.Tests/Edm/EdmModelExtensionsTests.cs:77]: Done.

File: test/Microsoft.OpenAPI.OData.Reader.Tests/Edm/EdmModelExtensionsTests.cs (hunk 4 above)
Comment (garethj-msft) [reply test/Microsoft.OpenAPI.OData.Reader.Tests/Edm/EdmModelExtensionsTests.cs:63]: Done.

**Pull request merged**